src/mono/browser/runtime/hybrid-globalization/locales.ts

```diff
@@ -75,42 +72,32 @@ export function mono_wasm_get_locale_info (culture: MonoStringRef, locale: MonoS
     } catch (ex: any) {
         setI32(dstLength, -1);
         return stringToUTF16Ptr((ex));
-    } finally {
-        cultureRoot.release();
     }
 }
 
-export function mono_wasm_get_first_day_of_week (culture: MonoStringRef, resultPtr: Int32Ptr): VoidPtr {
-
-    const cultureRoot = mono_wasm_new_external_root<MonoString>(culture);
+export function mono_wasm_get_first_day_of_week (culture: number, cultureLength: number, resultPtr: Int32Ptr): VoidPtr {
     try {
-        const cultureName = monoStringToString(cultureRoot);
+        const cultureName = utf16ToString(<any>culture, <any>(culture + 2 * cultureLength));
         const canonicalLocale = normalizeLocale(cultureName);
         const result = getFirstDayOfWeek(canonicalLocale);
         setI32(resultPtr, result);
         return VoidPtrNull;
     } catch (ex: any) {
         setI32(resultPtr, -1);
         return stringToUTF16Ptr((ex));
```

Comment:
`stringToUTF16Ptr` needs string, but ex is Error object. 
I think we need to call `toStrring()` on it or even better `normalize_exception`
And somehow test it, because this shows we don't have this covered.

Reply:
Tested and `ToString` is needed, otherwise we loose JS stack and return trash like `System.Exception : 촤 촤 `. Automated tests are not possible - C# libs are always expected to use the functions in the correct way, there's no API that fails in JS by design.

Reply:
are you saying that the errors are so unexpected that this should be abort instead ?
We could abort on JS side and it would make everything even simpler.

Reply:
We want to know what lib called JS to investigate the problem correctly, when we abort in JS then we loose this info. The exceptions will be useful when changing host or host version. As long as host behavior is stable (typical CI) then we cannot expect them to throw.